frontend/web/src/app/components/material/material-list/material-list.component.ts

```diff
@@ -0,0 +1,170 @@
+import {AfterViewInit, Component, EventEmitter, OnInit, Output, ViewChild} from '@angular/core';
+import { Router } from '@angular/router';
+import { BsModalRef, BsModalService } from 'ngx-bootstrap';
+import {Material} from '../../../model/material';
+import {ErrorService} from '../../../services/error.service';
+import {MaterialService} from '../../../services/material/material.service';
+
+declare var $: any;
+
+@Component({
+  selector: 'app-material-list',
+  templateUrl: './material-list.component.html',
+  styleUrls: ['./material-list.component.scss'],
+  providers: [MaterialService, ErrorService],
+})
+export class MaterialListComponent implements OnInit, AfterViewInit {
+  @Output() selected: EventEmitter<Material> = new EventEmitter<Material>();
+  private table;
+  private materials: Material[];
+
+  private modalRef: BsModalRef;
+  @ViewChild('modalDelete') modalDelete;
+  private toBeDeleted: number = null;
+  selectedMaterial: Material = null;
+
+  constructor(private materialService: MaterialService,
+              private errorService: ErrorService,
+              private router: Router,
+              private modalService: BsModalService) {
+  }
+
+  ngOnInit() {
+    this.table = $('#table');
+  }
+
+  ngAfterViewInit(): void {
+    this.loadAndPopulate();
+  }
+
+  loadAndPopulate() {
+    /* get users */
+    this.materialService.getMaterials().subscribe(
+      (materials) => {
+        this.materials = materials;
+        this.populate();
+        this.prepareTriggers();
+      }, (error) => {
+        alert(error.verbose_message);
+      },
+    );
+  }
+
+  private populate() {
+    const _self = this;
+    /*manually generate columns*/
+    const columns = [{
+      title: 'ID',
+      field: 'id',
+      sortable: true,
+    }, {
+      title: 'Name',
+      field: 'name',
+      sortable: true,
+    }, {
+      title: 'Supplier name',
+      field: 'supplierName',
+      sortable: true,
+    }, {
+      title: 'Initial amount',
+      field: 'initialAmount',
+      sortable: true,
+    }, {
+      field: 'operate',
+      title: '<span class="glyphicon glyphicon-cog">',
+      align: 'center',
+      events: {
+        'click .edit'(e, value, row, index) {
+          _self.router.navigate([_self.router.url, row.id]);
+        },
+        'click .delete'(e, value, row, index) {
+          _self.delete(row.id);
+        },
+      },
+      formatter: this.operateFormatter,
+    }];
+
+    const data = [];
+    this.materials.forEach((material) => {
+      data.push({
+        id: material.id,
+        name: material.name,
+        supplierName: material.supplierName,
+        initialAmount: material.initialAmount,
+        materialGrades: material.materialGrades,
+        materialProperties: material.materialProperties,
+      });
+    });
+
+    (this.table as any).bootstrapTable('destroy');
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4325)